client/src/components/payments/checkoutForm.js

```diff
@@ -0,0 +1,218 @@
+import { CardElement, injectStripe } from 'react-stripe-elements';
+import { stripeStyle } from './stripeStyles';
+import { pay } from '../../api/payments';
+import { toDollars } from '../../helpers/formatting';
+
+const styles = (muiBaseTheme) => ({
+    card: {
+        width: '500px',
+        margin: '50px auto',
+        padding: muiBaseTheme.spacing.unit * 3,
+        transition: '0.5s',
+        boxShadow: '0 8px 40px -12px rgba(0,0,0,0.3)',
+        '&:hover': {
+            boxShadow: '0 16px 70px -12.125px rgba(0,0,0,0.3)'
+        },
+        [muiBaseTheme.breakpoints.down('sm')]: {
+            width: '100%',
+            padding: "0"
+        }
+    },
+    content: {
+        textAlign: 'center',
+        padding: muiBaseTheme.spacing.unit
+    },
+    investmentTypo: {
+        marginBottom: '40px'
+    },
+    paymentTypo: {
+        textAlign: "center", margin: "15px 0 -10px"
+    },
+    divider: {
+        margin: '35px auto 25px'
+    },
+    paymentDivider: {
+        margin: '35px auto 25px',
+        color: 'white',
+        marginTop: '42px'
+    },
+    input: {
+        width: '100%',
+        height: '3.875em',
+    },
+    button: {
+        margin: '4px 0px 13px 0px'
+    },
+    outlinedInput: {
+        paddingLeft: "25px"
+    },
+    form: {
+        paddingBottom: "17px",
+        textAlign: "right",
+        marginRight: "8px"
+    }
+});
+
+class _CheckoutForm extends Component {
+
+    state = {
+        investmentAmount: "",
+        investmentSaved: false
+    }
+
+    handleInvestmentInput = (e) => {
+        const investmentAmount = e.target.value
+        if (investmentAmount <= 999999) {
+            this.setState({ investmentAmount })
+        }
+    }
+
+    handleInvestmentSubmit = (e) => {
+        e.preventDefault();
+        if (this.state.investmentAmount >= 5) {
+            this.setState({ investmentSaved: true })
+        } else {
+            this.props.activateToast('Sorry, $5 minimum', 'neutral')
+        }
+    }
+
+    handlePaymentSubmit = (e) => {
+        e.preventDefault();
+        // const { userId, projectId, stripe, activateToast } = this.props
+        const { userId, projectId, stripe, history, activateToast } = this.props
+        const { investmentAmount } = this.state
+
+        stripe.createToken().then((payload) => {
+            pay(projectId, payload, investmentAmount)
+                .then((investment) => investment)
+                .then(() => activateToast('success. you invested.', 'success'))
+                .then(() => history.push("/explore"))
+                .catch(() => activateToast('that was a fail', 'error'))
+        });
+    };
+
+    renderPaymentCard = (classes) => {
+        const { investmentAmount } = this.state
+
+        return (
+            <Fragment>
+                <CardContent className={classes.content}>
+                    <Typography
+                        className={`MuiTypography--heading ${classes.paymentTypo}`}
```

Comment:
I'm just curious as to why you put MuiTypography--heading in className, I would think it would already be there, no?

Reply:
because I needed different variables. Just let me know if there is a better way